### PR TITLE
Create release plugin

### LINKED ISSUE #28: Create a release plugin
The plugin should apply and configure the following plugins:
- net.researchgate.release
- org.hildan.github.changelog

Addtionally it should allow disabling push to remote through a property `disablePushToRemote` when releasing:
```groovy
project.extensions.disablePushToRemote = project.hasProperty("disablePushToRemote") ? project.disablePushToRemote : false
project.release {
    git {
        pushToRemote.set(project.disablePushToRemote.toBoolean() ? false : "origin")
    }
}
```

### PR DESCRIPTION
**Pull request by philipp94831**
Fixes #28

### CODE REVIEW

File: release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.configure
+
+class ReleasePlugin : Plugin<Project> {
+
+    companion object {
+        const val DISABLE_PUSH_TO_REMOTE = "release.disablePushToRemote"
+        const val REQUIRE_BRANCH = "release.requireBranch"
```
Comment (torbsto) [inline release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: Maybe we can make it an extension, then you don't have to deal with the type conversions and users can configure it in the build.gradle.kts.
```kotlin
interface ReleasePluginExtension {
    val disablePushToRemote: Property<Boolean>
    val requireBranch: Property<String?>
}

// ...
 val extension = rootProject.extensions.create<ReleasePluginExtension>("release")
```

Comment (philipp94831) [reply release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: Users should use the standard ReleasePluginExtension I think. See example here https://github.com/bakdata/streams-bootstrap/blob/master/build.gradle.kts#L86. These blocks just become unnecessary for our projects then

Comment (philipp94831) [reply release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: This plugin just adds the config options via properties at runtime

File: README.md [hunk 2]
```diff
@@ -7,8 +7,9 @@ bakdata gradle plugins
 
 A collection of small gradle plugin, mostly focused on deployment.
```
Comment (yannick-roeder) [inline README.md:8]:
```suggestion
A collection of small Gradle plugins, mostly focused on deployment.
```

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

**Review by yannick-roeder — APPROVED** [review]

File: release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt (hunk 1 above)
Comment (torbsto) [reply release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: do we know why `release.git.requireBranch` doesn't work?  We're already using `-Prelease.releaseVersion` for example

Comment (philipp94831) [reply release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: Not every extension config can be set via gradle properties. They query the releaseVersion manually https://github.com/researchgate/gradle-release/blob/main/src/main/groovy/net/researchgate/release/PluginHelper.groovy#L143

Comment (philipp94831) [reply release/src/main/kotlin/com/bakdata/gradle/ReleasePlugin.kt:38]: Using `release.git.requireBranch` did not work https://github.com/bakdata/github-playground/actions/runs/8139899744/job/22244025674#step:2:673 (job should have failed)

**Pull request merged**